public/docs/ts/latest/guide/reactive-forms.jade

```diff
@@ -0,0 +1,879 @@
+  The `setValue` method checks the data object thoroughly before assigning any form control values.
+  It rejects a data object that doesn't match the `FormGroup` structure.
+  It rejects a data object if it is missing values for any control in the group.
+  It does return helpful errors.
+
+  Notice that you can _almost_ use the entire `hero` as the argument to `setValue`
+  because its shape is similar to the component's `FormGroup` structure.
+
+  You can only show the hero's first address and you must account for the possibility that the `hero` has no addresses at all.
+  This explains the conditional setting of the `address` property in the data object argument:
++makeExample('reactive-forms/ts/app/hero-detail-versions.component.ts', 'set-value-address')(format=".")
+
+:marked
+  ### _patchValue_
+  With **`patchValue`**, you can assign values to specific controls in a `FormGroup`
+  by supplying an object of key/value pairs for just the controls of interest. 
+
+  This example sets only the form's `name` control.
++makeExample('reactive-forms/ts/app/hero-detail-versions.component.ts', 'patch-value','app/hero-detail.component.ts (excerpt)')(format=".")
+
+:marked
+  With **`patchValue`** you have more flexibility to cope with wildly divergent data and form models.
+  But unlike `setValue`,  `patchValue` cannot check for missing control values and does not return errors.
+
+  ### When to set form model values (_ngOnChanges_)
+
+  Now you know _how_ to set the _form model_ values. But _when_ do you set them?
+  The answer depends upon when the component gets the _data model_ values.
+  
+  If the `HeroDetailComponent` is editing a _new_ hero, the data model consists entirely of default values
+  that you could copy to the `FormControls` when you create them in the component's constructor.
+  That's effectively what you did when you set each control to the empty string.
+
+  If the `HeroDetailComponent` is editing an _existing_ hero, you have to wait for that hero to arrive.
+  The component _might_ have the ability to retrieve the hero from the server (via an injected data service)
+  in which case you could set the form model in the [ngOnInit](lifecyle-hooks.html#oninit) lifecycle hook.
+  The final version of the `HeroDetailComponent` in the [Tour of Heroes](../tutorial/toh-pt6.html) does it this way.
+
+  The `HeroDetailComponent` in this reactive forms sample works a little differently.
+  It's the _detail_ view nested within a _master/detail_ `HeroListComponent` ([discussed below](#hero-list)).
+  
+  The `HeroListComponent` displays hero names to the user.
+  When the user clicks on a hero, the list component passes the selected hero into the `HeroDetailComponent`
+  by **binding to the latter's `hero` input property**.
+
++makeExample('reactive-forms/ts/app/hero-list.component.1.html', '','hero-list.component.html (simplified)')(format=".")
+
+:marked
+  In this approach, the value of `hero` in the `HeroDetailComponent` changes 
```

Comment:
@wardbell As it is ... I think the way we are doing the imports all at the top give the reader the impression that Input/Output are required for reactive forms. They are not. They are only required here because of how we selected to build this app as a master/detail. I think the reader trying to skim this chapter to figure out how to build a reactive form is going to be confused by all of these extra imports here at the top.